ts/src/oceanex.ts

```diff
@@ -644,19 +648,30 @@ export default class oceanex extends Exchange {
          * @param {float} amount how much of currency you want to trade in units of base currency
          * @param {float} [price] the price at which the order is to be fullfilled, in units of the quote currency, ignored in market orders
          * @param {object} [params] extra parameters specific to the oceanex api endpoint
+         * @param {float} [params.triggerPrice] the price a trigger order is triggered at
+         *
+         * EXCHANGE SPECIFIC PARAMETERS
+         * @param {string} [params.trigger_condition] "gte" (default for trigger orders where side is "buy"), or "lte" (default for trigger orders where side is "sell")
          * @returns {object} an [order structure]{@link https://github.com/ccxt/ccxt/wiki/Manual#order-structure}
          */
         await this.loadMarkets ();
         const market = this.market (symbol);
```

Comment:
@samgermain can you place orders using the python version? I'm getting this weird error
```
p oceanex createOrder LTC/USDT limit sell 0.1 200 '{"stopPrice": 150}' --verbose  
Python v3.11.5
CCXT v4.0.109
oceanex.createOrder(LTC/USDT,limit,sell,0.1,200,{'stopPrice': 150})
Traceback (most recent call last):
  File "/Users/cjg/Git/ccxt10/ccxt/examples/py/cli.py", line 239, in <module>
    asyncio.run(main())
  File "/opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/cjg/Git/ccxt10/ccxt/examples/py/cli.py", line 216, in main
    result = await method(*args)
             ^^^^^^^^^^^^^^^^^^^
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/oceanex.py", line 634, in create_order
    response = await self.privatePostOrders(self.extend(request, params))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/base/exchange.py", line 1870, in request
    return await self.fetch2(path, api, method, params, headers, body, config)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/base/exchange.py", line 1866, in fetch2
    request = self.sign(path, api, method, params, headers, body)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/oceanex.py", line 908, in sign
    jwt_token = self.jwt(request, self.encode(self.secret), 'sha256', True)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/base/exchange.py", line 1303, in jwt
    signature = Exchange.base64_to_binary(Exchange.rsa(token, Exchange.decode(secret), algorithm))
                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/base/exchange.py", line 1316, in rsa
    priv_key = load_pem_private_key(Exchange.encode(secret), None, backends.default_backend())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/cryptography/hazmat/primitives/serialization/base.py", line 24, in load_pem_private_key
    return ossl.load_pem_private_key(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 949, in load_pem_private_key
    return self._load_key(
           ^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1201, in _load_key
    self._handle_key_loading_error()
  File "/opt/homebrew/lib/python3.11/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1257, in _handle_key_loading_error
    raise ValueError(
ValueError: ('Could not deserialize key data. The data may be in an incorrect format, it may be encrypted with an unsupported algorithm, or it may be an unsupported key type (e.g. EC curves with explicit parameters).', [_OpenSSLErrorWithText(code=503841036, lib=60, reason=524556, reason_text=b'error:1E08010C:DECODER routines::unsupported')])
```

Reply:
>  can you place orders using the python version? I'm getting this weird error

I think it has something to do with the weird format of the api key, because this is the only exchange that uses an RSA key

Reply:
@samgermain no, binance also supports rsa

Reply:
> @samgermain no, binance also supports rsa

but not in the same way, in that we literally create an rsa key using the terminal and copy and paste the rsa secret key into `keys.local.json`

Reply:
@samgermain but what's the difference if they're both rsa?

Reply:
Besides the signature issue, error handling seems off as well

```
n oceanex createOrder LTC/USDT limit sell 0.1 200 '{"stopPrice": 150}' --verbose
Debugger attached.
2023-10-26T09:39:16.662Z
Node.js: v18.18.0
CCXT v4.0.109
oceanex.createOrder (LTC/USDT, limit, sell, 0.1, 200, [object Object])
fetch Request:
 oceanex POST https://api.oceanex.pro/v1/orders?user_jwt=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1aWQiOiJJRDBBMzhGNTE5NjAiLCJkYXRhIjp7Im1hcmtldCI6Imx0Y3VzZHQiLCJzaWRlIjoic2VsbCIsIm9yZF90eXBlIjoic3RvcF9saW1pdCIsInZvbHVtZSI6IjAuMSIsInByaWNlIjoiMjAwIiwic3RvcF9wcmljZSI6IjE1MCIsInRyaWdnZXJfY29uZGl0aW9uIjoibHRlIn19.fWwjq8-ccMbHSldxCwNjCS5AKKWTIaP3XSVYd8xomWW1UUnbEsC_4ROJDsC4BxqSh8aZ_SsThC2Wq4iSPIjFyxfvyWm_NtJGEMSAcV20ohMz_iHQ1NrFshg-X065DAVu6U7z3anL-23eTmbpWik4phSzU0dnpiWAfsKMv_APnXlFsF7FVwIctbtDX4t9ZWdYuc6pBPcvpN6JFfmun6_jRyFGze8g-FfVjG30OLEwxGhDe31ibI-vyQI9mdZgWU92wykpBVscPP8vmTiMIKSGDK5OnOTiUGA1Bu3oEFMq3U1OWRsEwwwdcDRsdAvghsFA7l3rTuEqe40lcmrviNzF3A 
RequestHeaders:
 { 'Content-Type': 'application/json' } 
RequestBody:
 undefined 

handleRestResponse:
 oceanex POST https://api.oceanex.pro/v1/orders?user_jwt=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1aWQiOiJJRDBBMzhGNTE5NjAiLCJkYXRhIjp7Im1hcmtldCI6Imx0Y3VzZHQiLCJzaWRlIjoic2VsbCIsIm9yZF90eXBlIjoic3RvcF9saW1pdCIsInZvbHVtZSI6IjAuMSIsInByaWNlIjoiMjAwIiwic3RvcF9wcmljZSI6IjE1MCIsInRyaWdnZXJfY29uZGl0aW9uIjoibHRlIn19.fWwjq8-ccMbHSldxCwNjCS5AKKWTIaP3XSVYd8xomWW1UUnbEsC_4ROJDsC4BxqSh8aZ_SsThC2Wq4iSPIjFyxfvyWm_NtJGEMSAcV20ohMz_iHQ1NrFshg-X065DAVu6U7z3anL-23eTmbpWik4phSzU0dnpiWAfsKMv_APnXlFsF7FVwIctbtDX4t9ZWdYuc6pBPcvpN6JFfmun6_jRyFGze8g-FfVjG30OLEwxGhDe31ibI-vyQI9mdZgWU92wykpBVscPP8vmTiMIKSGDK5OnOTiUGA1Bu3oEFMq3U1OWRsEwwwdcDRsdAvghsFA7l3rTuEqe40lcmrviNzF3A 200 OK 
ResponseHeaders:
 {
  'Access-Control-Allow-Credentials': 'true',
  'Access-Control-Allow-Headers': 'Origin, X-Requested-With, Content-Type, Accept, Authorization',
  'Access-Control-Allow-Methods': 'GET, POST, PUT, PATCH, DELETE',
  'Access-Control-Allow-Origin': 'https://oceanex.pro',
  'Cache-Control': 'max-age=0, private, must-revalidate',
  'Cf-Cache-Status': 'DYNAMIC',
  'Cf-Ray': '81c1bb45285563f9-LHR',
  Connection: 'keep-alive',
  'Content-Encoding': 'br',
  'Content-Security-Policy': "default-src 'self' storage.googleapis.com;style-src 'self' 'unsafe-inline';style-src-elem 'self' 'unsafe-inline' translate.googleapis.com fonts.googleapis.com;script-src 'self' 'unsafe-inline' 'unsafe-eval' *.identitymind.com *.cdnwidget.com *.cdnbasket.net *.polyfill.io polyfill.io *.google-analytics.com *.pusher.com *.captcha.qq.com *.recaptcha.net recaptcha.net www.gstatic.com *.gstatic.cn www.google-analytics.com *.googletagmanager.com static.cloudflareinsights.com captcha.gtimg.com apis.google.com;script-src-elem 'self' 'unsafe-inline' *.oceanex.pro *.identitymind.com *.polyfill.io polyfill.io *.cdnwidget.com www.gstatic.com *.recaptcha.net recaptcha.net *.captcha.qq.com www.google.com *.google-analytics.com *.pusher.com *.gstatic.cn translate.googleapis.com translate.google.com *.googletagmanager.com static.cloudflareinsights.com captcha.gtimg.com apis.google.com;font-src 'self' data: *;img-src data: blob: 'self' *;form-action 'self' checkout.simplexcc.com;connect-src 'self' *.oceanex.pro *.cdnwidget.com ws://ws-slanger.oceanex.pro wss://ws-slanger.oceanex.pro wss://ws-slanger.oceanex.pro:443  *.cdnbasket.net *.sentry.io *.google-analytics.com *.pusher.com *.doubleclick.net sentry.io translate.googleapis.com storage.googleapis.com devapi.bbx.com api.bbx.com ws://api.bbx.com wss://api.bbx.com *.tradingview.com *.tiger.cloud wss://api.tiger.cloud *.google.com;frame-ancestors 'self';frame-src 'self' *.captcha.qq.com www.google.com recaptcha.net accounts.google.com;object-src 'none';report-uri https://sentry.io/api/1504690/security/?sentry_key=98071b02061b46fa9a78c4960265ecc5",
  'Content-Type': 'application/json',
  Date: 'Thu, 26 Oct 2023 09:39:20 GMT',
  Etag: 'W/"b987277c5360a5c432274440fb76f602"',
  Nel: '{"success_fraction":0,"report_to":"cf-nel","max_age":604800}',
  'Report-To': '{"endpoints":[{"url":"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=Swj5iByu%2FPM3p%2FXyGaRWWXJfvXsP5y9rshSmkdhr4n%2FyNJA%2BKOvUH8JJsu%2BRMYKMnU6Roe0xd5iifnyVoJw6WhMEnn7%2B5M%2F9JY0KVhnYM2QtAZtVq99iBuEYhZEFaJbrLg%3D%3D"}],"group":"cf-nel","max_age":604800}',
  Server: 'cloudflare',
  'Strict-Transport-Security': 'max-age=15724800; includeSubDomains',
  'Transfer-Encoding': 'chunked',
  Vary: 'Origin',
  'X-Cloud-Trace-Context': 'ca55ba95d8f041468ed36b9609091b20;o=0',
  'X-Frame-Options': 'SAMEORIGIN',
  'X-Request-Id': 'e9bdbf7fed4cc89c7daf3bdda61c3665',
  'X-Robots-Tag': 'noindex',
  'X-Runtime': '0.084473'
} 
ResponseBody:
 {"code":-2,"message":"{\"error\":{\"code\":2002,\"message\":\"Failed to create order. #\\u003cAccount::AccountError: Cannot lock funds (amount: 0.1).\\u003e\"}}","data":{}} 

2023-10-26T09:39:20.894Z iteration 0 passed in 3479 ms

{
  info: undefined,
  id: undefined,
  clientOrderId: undefined,
  timestamp: undefined,
  datetime: undefined,
  lastTradeTimestamp: undefined,
  symbol: 'LTC/USDT',
  type: undefined,
  timeInForce: undefined,
  postOnly: undefined,
  side: undefined,
  price: undefined,
  stopPrice: undefined,
  triggerPrice: undefined,
  average: undefined,
  amount: undefined,
```

Reply:
> Besides the signature issue, error handling seems off as well
> 

This is fixed in the most recent commit

```
% oceanex createOrder LTC/USDT limit sell 0.1 200 '{"stopPrice": 150}'
2023-11-10T06:04:44.458Z
Node.js: v18.15.0
CCXT v4.0.109
oceanex.createOrder (LTC/USDT, limit, sell, 0.1, 200, [object Object])
InvalidOrder oceanex {"code":-2,"message":"{\"error\":{\"code\":2002,\"message\":\"Failed to create order. #\\u003cAccount::AccountError: Cannot lock funds (amount: 0.1).\\u003e\"}}","data":{}}
---------------------------------------------------
[InvalidOrder] oceanex {"code":-2,"message":"{\"error\":{\"code\":2002,\"message\":\"Failed to create order. #\\u003cAccount::AccountError: Cannot lock funds (amount: 0.1).\\u003e\"}}","data":{}}

    at throwExactlyMatchedException  …samgermain/Documents/CCXT/ccxt/ts/src/base/Exchange.ts:3506  
    at handleErrors                  Users/samgermain/Documents/CCXT/ccxt/ts/src/oceanex.ts:1021   
    at                               …samgermain/Documents/CCXT/ccxt/ts/src/base/Exchange.ts:1016  
    at processTicksAndRejections     node:internal/process/task_queues:95                          
    at fetch2                        …samgermain/Documents/CCXT/ccxt/ts/src/base/Exchange.ts:3055  
    at request                       …samgermain/Documents/CCXT/ccxt/ts/src/base/Exchange.ts:3059  
    at createOrder                   Users/samgermain/Documents/CCXT/ccxt/ts/src/oceanex.ts:676    
    at async run                     Users/samgermain/Documents/CCXT/ccxt/examples/ts/cli.ts:301   

oceanex {"code":-2,"message":"{\"error\":{\"code\":2002,\"message\":\"Failed to create order. #\\u003cAccount::AccountError: Cannot lock funds (amount: 0.1).\\u003e\"}}","data":{}}
```

Reply:
> @samgermain but what's the difference if they're both rsa?

I don't know, I don't have that deep of an understanding of RSA or how were deserializing it, but it looks like the key is probably the problem based on the error message `ValueError: ('Could not deserialize key data. The data may be in an incorrect format, it may be encrypted with an unsupported algorithm, or it may be an unsupported key type (e.g. EC curves with explicit parameters).', [_OpenSSLErrorWithText(code=503841036, lib=60, reason=524556, reason_text=b'error:1E08010C:DECODER routines::unsupported')])`

Reply:
@carlosmiei Would the problem have anything to do with the oceanex keys starting with `-----BEGIN RSA PRIVATE KEY-----`, there's spaces in there, which aren't something that's present in other secret keys